File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.cpp
class RandomizedSet {
public:
    unordered_set<int>st;
    RandomizedSet() {
        
    }
    
    bool insert(int val) {
        int prevSz = st.size();
        st.insert(val);
        return !(prevSz == st.size());
    }
    
    bool remove(int val) {
        return st.erase(val);   
    }
    
    int getRandom() {
        int idx = (rand() % st.size());
        
        int res = -1;
        
        for(int i : st){
            res = i;
            --idx;
            if(idx == 0)    break;
        }
        
        return res;
    }
};

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet* obj = new RandomizedSet();
 * bool param_1 = obj->insert(val);
 * bool param_2 = obj->remove(val);
 * int param_3 = obj->getRandom();
 */

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.java
class RandomizedSet {
    Set<Integer>set;
    public RandomizedSet() {
        set = new HashSet<>();
    }
    
    public boolean insert(int val) {
        return set.add(val);
    }
    
    public boolean remove(int val) {
        return set.remove(val);
    }
    
    public int getRandom() {
        int sz = set.size();        
        int idx = new Random().nextInt(sz);
        
        Iterator<Integer>it = set.iterator();
        
        for(int i = 0; i < idx; i++){
            it.next();
        }
        
        return it.next();
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */